lib/cracker.rb

```diff
@@ -1,3 +1,61 @@
-class Cracker
+require './lib/decoder'
 
+class Cracker < Decoder
+
+  def reverse_shifts(ciphertext, known_end = " end")
+    last_four = ciphertext[-known_end.length..-1].split('')
+    last_four.zip(known_end.split('')).map do |cipher_char, known_char|
```

Comment:
this is basically pairing the last four characters of the ciphertext with their matching characters from " end". "abcd" turns into something like [["a", " "]["b", "e"]["c", "n"]["d", "d"]]